### ISSUE TITLE
Support signing firmware

### ISSUE THREAD

**Issue by lulf** [lulf/drogue-ajour#5]

**Comment by lulf** [comment]
Ideally this should use https://www.sigstore.dev/  

CC @bobmcwhirter

**Comment by danbev** [comment]
@lulf I've written some [notes](https://github.com/danbev/learning-crypto/blob/main/notes/firmware-sign-verifiy.md) for this task with some initial investigation and wondering if you be able to take a look and see if I'm on the right track with this?

**Comment by lulf** [comment]
@danbev It's a different approach to what I imagined when looking at ORAS, because one of the examples there is that you attach the signatures as a generic 'metadata' associated with the container image, instead of within the artifact itself. Bundling it as you suggest is an interesting approach as well, maybe we need to think about the pros/cons of each of those and decide if we support both or one of them, or maybe even a combination?

**Comment by danbev** [comment]
@lulf Thanks for the feedback! I'll take a closer look at the ORAS example today :+1:

**Comment by danbev** [comment]
@lulf I've taken a look at using [attachments](https://github.com/danbev/learning-crypto/blob/main/notes/firmware-sign-verifiy.md#attaching-the-bundle-to-the-image) and the issue I ran into previously was fixed with the latest update of oras (details are in the section linked). Is that what you hand in mind with regards to attachments?

**Comment by lulf** [comment]
@danbev Yes, I think the signature is a 'layer' with a media type and/or signature type. I think the example I looked at was this https://oras.land/cli/6_reference_types/

**Comment by danbev** [comment]
> I think the example I looked at was this https://oras.land/cli/6_reference_types/

Ah thanks, I'll read through that :+1:

**Comment by danbev** [comment]
@lulf 
I've added a section about using [references](https://github.com/danbev/learning-crypto/blob/main/notes/firmware-sign-verifiy.md#adding-the-bundle-as-a-reference) to try to understand how that works.

I've also added an [alternatives](https://github.com/danbev/learning-crypto/blob/main/notes/firmware-sign-verifiy.md#alternatives) section which sums up the alternative I think we have.
Do these [tasks](https://github.com/danbev/learning-crypto/blob/main/notes/firmware-sign-verifiy.md#tasks) seem reasonable to you?
